src/Magnum/Math/Geometry/Intersection.h

```diff
@@ -140,20 +142,246 @@ points distance from the plane is negative) using @ref Distance::pointPlaneScale
+
+/**
+@brief Intersection of a sphere and a cone
+@param sphereCenter Sphere center
+@param radius Sphere radius
+@param origin Origin of the cone
+@param normal Cone normal
+@param angle Cone opening angle (@f$ 0 < \text{angle} < \pi @f$).
+
+Returns @cpp true @ce if the sphere intersects with the cone.
+
+Offsets the cone plane by @f$ -r\sin{\theta} \cdot \boldsymbol n @f$ (with  @f$ \theta @f$
+the cone's half-angle) which separates two half-spaces:
+In front of the plane, in which the sphere cone intersection test is equivalent
+to testing the sphere's center against a similarly offset cone (which is equivalent
+the cone with surface expanded by @f$ r @f$ in surface normal direction), and
+behind the plane, where the test is equivalent to testing whether the origin of
+the original cone intersects the sphere.
+
+@see @ref sphereCone(const Vector3<T>& sphereCenter, T, const Vector3<T>&, const Vector3<T>&, T, T)
+*/
+template<class T> bool sphereCone(const Vector3<T>& sphereCenter, T radius, const Vector3<T>& origin, const Vector3<T>& normal, Rad<T> angle);
+
+/**
+@brief Intersection of a sphere and a cone using precomputed values
+@param sphereCenter Sphere center
+@param radius Sphere radius
+@param origin Origin of the cone
+@param normal Cone normal
+@param sinAngle Precomputed sine of half the cone's opening angle: @cpp Math::sin(angle/T(2)) @ce
+@param tanAngleSqPlusOne Precomputed portion of the cone intersection equation:
+                         @cpp Math::pow(Math::tan(angle/T(2)), 2) + 1 @ce
+
+Returns @cpp true @ce if the sphere intersects with the cone.
+*/
+template<class T> bool sphereCone(const Vector3<T>& sphereCenter, T radius, const Vector3<T>& origin, const Vector3<T>& normal, T sinAngle, T tanAngleSqPlusOne);
+
+/**
+@brief Intersection of an axis aligned bounding box and a cone
+
+@param center Center of the AABB
+@param extents (Half-)extents of the AABB
+@param origin Origin of the cone
+@param normal Normal of the cone
+@param angle Opening angle of the cone
+
+Returns @cpp true @ce if the axis aligned bounding box intersects the cone.
+
+On each axis finds the intersection points of the cone's axis with infinite planes
+obtained by extending the two faces of the box that are perpendicular to that axis.
+
+The intersection points on the planes perpendicular to axis @f$ a \in {0, 1, 2} @f$
+are given by @f[
+    \boldsymbol i = \boldsymbol n \cdot \frac{(\boldsymbol c_a - \boldsymbol o_a) \pm \boldsymbol e_a}{\boldsymbol n_a}
+@f]
+
+with normal @f$ n @f$, cone origin @f$ o @f$, box center @f$ x @f$ and box extents @f$ e @f$.
+
+The points on the faces that are closest to this intersection point are the closest
+to the cone's axis and are tested for intersection with the cone using
+@ref pointCone(const Vector3<T>&, const Vector3<T>&, const Vector3<T>&, const T) "pointCone".
```

Comment:
... `"pointCone()"`? I.e., so it's clear it refers to a function. 

Similarly in other cases.